### ISSUE TITLE
Module pip fails by default

### ISSUE THREAD

**Issue by ThomasSteinbach** [ansible/ansible#10810]
## Problem:

Running the pip module with e.g. following command...

```
- name: install python-docker
  pip: name=docker-py state=present
```

... works on first execution (probably after installing python-pip), but fails on later executions (maybe 24hrs later) with following error:

```
ImportError: cannot import name IncompleteRead
```
## Workaround:

A current workaround is to execute prior to the pip action:

```
- name: Make sure pip upgraded
  command: easy_install -U pip
```
## Solution proposal:

The pip module should offer an upgrade command like the apt module does with `update_cache=yes`.
## References:

Other sources where this problem occures (I found searching for a solution):
- https://github.com/anarcher/ansible-docker-registry/issues/1
- https://github.com/bobbyrenwick/ansible-pip/issues/6

**Comment by bcoca** [comment]
So this does not seem to be an ansible problem, pip is required for the pip module to work.  Your environment seems to lack or remove pip or have a corrupt installation of it for some reason. 

Closing this as ansible does deal with this correctly and shows a correct error message and a copy of the exception pip had on execution.

**Comment by ThomasSteinbach** [comment]
This is indeed a known bug of python-pip:
https://bugs.launchpad.net/ubuntu/+source/python-pip/+bug/1306991

This is the more annoying, it is shipped with Ubuntu 14.04.2. However I think it is OK to run the workaround just once, which should fix python-pip and avoid Ansible change messages in consecutive runs of the playbook.
